nexus/db-queries/src/db/queries/sled_reservation.rs

```diff
@@ -57,18 +57,70 @@ pub fn sled_find_targets_query(
                 COALESCE(SUM(CAST(sled_resource_vmm.reservoir_ram AS INT8)), 0) + "
             ).param().sql(" <= sled.reservoir_size
         ),
-        our_aa_groups AS (
+        our_a_groups AS (
+            SELECT group_id
+            FROM affinity_group_instance_membership
+            WHERE instance_id = ").param().sql("
+        ),
+        other_a_instances AS (
+            SELECT affinity_group_instance_membership.group_id,instance_id
+            FROM affinity_group_instance_membership
+            JOIN our_a_groups
+            ON affinity_group_instance_membership.group_id = our_a_groups.group_id
+            WHERE instance_id != ").param().sql("
+        ),
+        our_direct_aa_groups AS (
+            -- Anti-affinity groups to which our instance belongs
             SELECT group_id
             FROM anti_affinity_group_instance_membership
             WHERE instance_id = ").param().sql("
         ),
-        other_aa_instances AS (
+        other_direct_aa_instances AS (
             SELECT anti_affinity_group_instance_membership.group_id,instance_id
             FROM anti_affinity_group_instance_membership
-            JOIN our_aa_groups
-            ON anti_affinity_group_instance_membership.group_id = our_aa_groups.group_id
+            JOIN our_direct_aa_groups
+            ON anti_affinity_group_instance_membership.group_id = our_direct_aa_groups.group_id
             WHERE instance_id != ").param().sql("
         ),
+        our_indirect_aa_groups AS (
+            -- Anti-affinity groups to which our instance's affinity groups belong
+            SELECT
+                anti_affinity_group_id,
+                affinity_group_id,
+                CASE
+                    WHEN COUNT(*) OVER (PARTITION BY anti_affinity_group_id) = 1 THEN TRUE
+                    ELSE FALSE
+                END AS exactly_one_affinity_group
+            FROM anti_affinity_group_affinity_membership
+            WHERE affinity_group_id IN (SELECT group_id FROM our_a_groups)
+        ),
+        other_indirect_aa_instances_via_instances AS (
```

Comment:
These are instances we're "anti-affine" with, due to their **instance** membership in the anti-affinity group and also due to **our** membership in an affinity group (see: `our_indirect_aa_groups`).

|Our Instance | Other Instances |
|-- | -- |
| Member in affinity group belonging to anti-affinity group | Member in anti-affinity group |